File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        //Bruth force: 2 for => N2
        //sort?
        //hashmap
        let map = new Map()
        for (let i = 0; i < nums.length; i++) {
            let differ = target - nums[i]
            if (map.has(differ)) {
                return [map.get(differ), i]
            }   //has return key exist
            map.set(nums[i], i)
        }
        return []
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # solution 1: for loop 2 vòng để get ra phần tử cần
        # (On^2)
        n = len(numbers)
        for i in range(n):
            for j in range(i+1, n):
                if numbers[i] + numbers[j] == target:
                    return [i + 1, j + 1] 
        
        # solution 2: hashmap

        # solution 3: using bubble sort

        # solution 4: 2 pointer
        
